modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CreationMenu.js

```diff
@@ -29,74 +29,34 @@ function CreationMenu({inEmptyState, primaryItems}) {
 
 	const [active, setActive] = useState(false);
 
-	const labelNew = Liferay.Language.get('new');
-	const buttonLabel = primaryItems[0].label ?? labelNew;
+	const CreationMenuButton = ({label}) => {
```

Comment:
We should avoid using an inner component, as leads to bad performance and bugs. See more: https://react.dev/learn/your-first-component#nesting-and-organizing-components. Please add any new component in the root of the file. Since this implies forwarding all props, you may consider reducing it in scope. If you're leaving it as it is in latest version of PR, naming is a bit off: `CreationMenu` prefix is not needed as this is an internal component, `Button` suffix does not describe is correctly, at is includes more than a button.

Reply:
I see, thank you, in this case I think I should take a step back, it doesn't seem to be worth to pass all the props to a new component just to avoid a few verifications.